provider/ansible/gcp_utils.py

```diff
@@ -316,3 +310,25 @@ def _compare_value(self, value1, value2):
             pass
 
         return diff
+
+    def _compare_boolean(self, value1, value2):
+        try:
+            # Both True
+            if value1 and value2:
```

Comment:
It looks like value1 is always a bool (`elif isinstance(value1, bool)`) and value2 may or may not be a bool. If my understanding is correct, then this expression will evaluate to true if `value1 == True` and `value2 == 'false'`